provider/balance_checker_test.go

```diff
@@ -146,5 +232,43 @@ func TestBalanceCheckerStartsWithdrawal(t *testing.T) {
 		t.Fatal("timed out waiting for completion")
 	}
 
-	testScaffold.queryClient.AssertExpectations(t)
+	testScaffold.qc.AssertExpectations(t)
+}
+
+func TestBalanceCheckerMonitorsFunds(t *testing.T) {
+	testScaffold, bc := leaseMonitorForTest(t, 1, time.Second*100)
+	defer testScaffold.testBus.Close()
+
+	subscriber, err := testScaffold.testBus.Subscribe()
+	require.NoError(t, err)
+	defer subscriber.Close()
+
+	lid := mtypes.LeaseID{
+		Owner:    bc.ownAddr.String(),
+		DSeq:     1,
+		GSeq:     0,
+		OSeq:     0,
+		Provider: bc.ownAddr.String(),
+	}
+
+	time.Sleep(time.Second)
+
+	err = testScaffold.testBus.Publish(event.LeaseAddFundsMonitor{LeaseID: lid})
+	require.NoError(t, err)
+
+	select {
+	case msg := <-testScaffold.broadcast.txCh:
+		switch msg.(type) {
+		case *mtypes.MsgCloseBid:
+		default:
+			t.Errorf("received unexpected message")
```

Comment:
I think this is meant to say "did not receive expected message" since it is in a default case which gets hit if the type does not match the other cases.

Reply:
so message has been received but not of the type we expected.
`did not receive expected message` fits more outside of the switch.

`require.IsType` is more suitable anyways